src/java/org/apache/cassandra/db/memtable/pmem/PmemPartition.java

```diff
@@ -626,6 +616,158 @@ public Iterator<RangeTombstone> getMergedRangeTombstones(PartitionUpdate update)
         return ranges.iterator();
     }
 
+    /**
+     * Removes Tombstones which are older than gcGraceSeconds
+     *
+     * @return Expired Tombstone count
+     */
+    public long vaccumTombstones()
+    {
+        long removedTombstones = 0;
+
+        if (block.getLong(DELETION_INFO_OFFSET) > 0)
+        {
+            TransactionalMemoryBlock deletedPartitionBlock = heap.memoryBlockFromHandle(block.getLong(DELETION_INFO_OFFSET));
+            DeletionTime deletionTime = getPartitionDeletionTime(deletedPartitionBlock);
+            int gcBefore = getGcBefore(metadata(), FBUtilities.nowInSeconds());
+            if (deletionTime != null && deletionTime.localDeletionTime() < gcBefore)
+            {
+                deletedPartitionBlock.free();
+                block.setLong(DELETION_INFO_OFFSET, 0);
+            }
+        }
+        removedTombstones += removeExpiredRangeTombstone();
+        removedTombstones += removeExpiredDeletedRows();
+        return removedTombstones;
+    }
+
+    //Removes Row Tombstones which are older then gcgraceseconds
+    private int removeExpiredDeletedRows()
+    {
+        Iterator<LongART.Entry> rowItr = pmemRowMap.getRowMapTree().getEntryIterator();
+        final ArrayList<byte[]> artEntryKeyArray = new ArrayList<>();
+        while (rowItr.hasNext())
+        {
+            LongART.Entry nextEntry = rowItr.next();
+            Unfiltered unfiltered = getUnfiltered(nextEntry, true);
+            if (unfiltered != null && isExpired(pmemTableInfo, unfiltered))
+            {
+                artEntryKeyArray.add(nextEntry.getKey());
+            }
+        }
+        if (!artEntryKeyArray.isEmpty())
+        {
+            for (byte[] artEntryKey : artEntryKeyArray)
+            {
+                pmemRowMap.getRowMapTree().remove(artEntryKey, (Long rowhandle) -> {
+                    TransactionalMemoryBlock rowMemoryBlock = heap.memoryBlockFromHandle(rowhandle);
+                    rowMemoryBlock.free();
+                });
+            }
+        }
+
+        return artEntryKeyArray.size();
+    }
+
+    //Removes RangeTombstones which are older then gcgraceseconds
+    private int removeExpiredRangeTombstone()
```

Comment:
Isn't this a duplicate of the code above with a different iterator? Can we extract the common code?

Nit: my personal preference is to use a reference to a different method instead of the boolean (e.g. `removeExpired(pmemRtmMap.getRowMapTree().getEntryIterator(), this::getRow)` and `removeExpired(pmemRtmMap.getRangeTombstoneMarkerTree().getEntryIterator(),  this::getTombstoneMarker)`).